src/dashboards/components/AutoRefreshInput.tsx

```diff
@@ -0,0 +1,100 @@
+import React, {FC} from 'react'
+import {useDispatch, useSelector} from 'react-redux'
+
+// Components
+import DurationInput from 'src/shared/components/DurationInput'
+import {ButtonGroup} from '@influxdata/clockface'
+// Actions
+import {showOverlay, dismissOverlay} from 'src/overlays/actions/overlays'
+
+// Types
+import {AutoRefreshStatus} from 'src/types'
+
+// Actions
+import {
+  setAutoRefreshInterval,
+  setAutoRefreshStatus,
+  resetDashboardAutoRefresh,
+  setAutoRefreshInputValue,
+} from 'src/shared/actions/autoRefresh'
+
+// Selectors
+import {getCurrentDashboardId} from 'src/dashboards/selectors'
+import {getAutoRefreshForDashboard} from 'src/shared/selectors/autoRefresh'
+
+const SUGGESTIONS = [
+  'None',
+  '10s',
+  '15s',
+  '30s',
+  '1m',
+  '5m',
+  '10m',
+  '30m',
+  '1h',
+]
+
+const AutoRefreshInput: FC = () => {
+  const dispatch = useDispatch()
+  const currentDashboardId = useSelector(getCurrentDashboardId)
+
+  const autoRefresh = useSelector(getAutoRefreshForDashboard)
+  const handleChooseAutoRefresh = (selection: string) => {
+    if (selection === 'None') {
+      dispatch(resetDashboardAutoRefresh(currentDashboardId))
+      return
+    }
+    const unit = selection[selection.length - 1]
+    let multiplier
+    if (unit === 's') {
+      multiplier = 1
+    } else if (unit === 'm') {
+      multiplier = 60
+    } else if (unit === 'h') {
+      multiplier = 3600
+    }
+
+    const calculatedMilliseconds =
+      Number(selection.slice(0, selection.length - 1)) * multiplier * 1000
```

Comment:
what if the `multiplier` is undefined

Reply:
Great catch, thanks, that should be inside of the if

Reply:
that would be fine if the only thing we needed to know is that its defined, but its not - we also need to know that it is one of the three units i listed